test/suite/test_timestamp12.py

```diff
@@ -0,0 +1,119 @@
+
+import shutil, os, wiredtiger, wttest
+from wtscenario import make_scenarios
+
+def timestamp_str(t):
+    return '%x' % t
+
+class test_timestamp12(wttest.WiredTigerTestCase):
+    conn_config = 'config_base=false,create,log=(enabled)'
+    coll_uri = 'table:collection12'
+    oplog_uri = 'table:oplog12'
+    closecfg = [
+        ('dfl', dict(close_cfg='', all_expected=False)),
+        ('use_stable', dict(close_cfg='use_timestamp=true', all_expected=False)),
+        ('all_dirty', dict(close_cfg='use_timestamp=false', all_expected=True)),
+   ]
+    scenarios = make_scenarios(closecfg)
+
+    def verify_expected(self, op_exp, coll_exp):
+        c_op = self.session.open_cursor(self.oplog_uri)
+        c_coll = self.session.open_cursor(self.coll_uri)
+        op_actual = dict((k, v) for k, v in c_op if v != 0)
+        coll_actual = dict((k, v) for k, v in c_coll if v != 0)
+        #print "CHECK: Op Expected"
+        #print op_exp
+        #print "CHECK: Op Actual"
+        #print op_actual
+        self.assertTrue(op_actual == op_exp)
+        #print "CHECK: Coll Expected"
+        #print coll_exp
+        #print "CHECK: Coll Actual"
+        #print coll_actual
+        self.assertTrue(coll_actual == coll_exp)
+
+    def test_timestamp_recovery(self):
+        if not wiredtiger.timestamp_build():
+            self.skipTest('requires a timestamp build')
+
+        #
+        # Create several collection-like tables that are checkpoint durability.
+        # Add data to each of them separately and checkpoint so that each one
+        # has a different stable timestamp.
+        #
+        self.session.create(self.oplog_uri, 'key_format=i,value_format=i')
+        self.session.create(self.coll_uri, 'key_format=i,value_format=i,log=(enabled=false)')
+        c_op = self.session.open_cursor(self.oplog_uri)
+        c_coll = self.session.open_cursor(self.coll_uri)
+
+        # Begin by adding some data.
+        nentries = 10
+        first_range = range(1, nentries)
+        second_range = range(nentries, nentries*2)
+        all_keys = range(1, nentries*2)
+        for i in first_range:
+            self.session.begin_transaction()
+            c_op[i] = 1
+            c_coll[i] = 1
+            self.session.commit_transaction(
+              'commit_timestamp=' + timestamp_str(i))
+        # Set the oldest and stable timestamp to the end.
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(nentries-1) +
+        ',stable_timestamp=' + timestamp_str(nentries-1))
+
+        # Add more data but don't advance the stable timestamp.
+        for i in second_range:
+            self.session.begin_transaction()
+            c_op[i] = 1
+            c_coll[i] = 1
+            self.pr("i: " + str(i))
+            self.session.commit_transaction(
+              'commit_timestamp=' + timestamp_str(i))
+
+        # Close and reopen the connection.
+        self.close_conn(self.close_cfg)
+        self.conn = self.setUpConnectionOpen(".")
```

Comment:
@sueloverso, I'd change these two lines (starting with `self.conn = `) to just:
```
self.open_conn()
```

Reply:
Fixed.